src/main/java/seedu/address/model/AutoRescheduleManager.java

```diff
@@ -0,0 +1,111 @@
+                Event event = item.getEvent().get();
+                if (event.hasAutoReschedule()) {
+                    if (event.getStartDateTime().compareTo(LocalDateTime.now()) > 0) {
+                        // event date is after now
+                        // add event to Timer thread, add(newEvent)
+                        RescheduleTask task = new RescheduleTask(item, event.getPeriod(), model);
+                        getInstance().add(task);
+                    } else {
+                        // event date is before now, but is reschedulable
+                        // modify the event date to the most upcoming one (Use modulo and add remainder)
+                        // add(newEvent)
+                        LocalDateTime updatedDateTime = getUpdatedDateTime(event);
+                        Event updatedEvent = event.changeStartDateTime(updatedDateTime);
+
+                        // update the old event time to the new one in the itemModel
+                        Item oldItem = item;
+                        Item newItem = item.changeEvent(updatedEvent);
+                        RescheduleTask task = new RescheduleTask(newItem, updatedEvent.getPeriod(), model);
+                        model.replaceItem(oldItem, newItem);
+
+                        getInstance().add(task);
+                    }
+
+                }
+            }
+        }
+    }
+
+    /**
+     * Update the old event start time to the closest upcoming start time, using the event's auto-reschedule period
+     * @param event whose start date we want to modify
+     * @return LocalDateTime representation of the modified time
+     */
+    private static LocalDateTime getUpdatedDateTime(Event event) {
+        // Use modulo to get the remaining time till the next reschedule time. Add that remaining time to the time now.
+        long period = event.getPeriod().getPeriod();
+        LocalDateTime startDateTime = event.getStartDateTime();
+        long millisDifference = Duration.between(startDateTime, LocalDateTime.now()).toMillis(); // positive difference;
+        BigInteger periodBi = new BigInteger("" + period);
+        BigInteger millisDifferenceBi = new BigInteger("" + millisDifference);
+
+        long millisRemainder = millisDifferenceBi.mod(periodBi).longValue();
+        LocalDateTime updatedDateTime = LocalDateTime.now().plusNanos(Duration.ofMillis(millisRemainder).toNanos());
+
+        return updatedDateTime;
+    }
+
+    /**
+     * Adds a RescheduleTask that is to be rescheduled periodically, to this AutoRescheduleManager.
+     * @param task RescheduleTask to be carried out after reaching an event's startDateTime
+     */
+    public void add(RescheduleTask task) {
+        try {
+            Duration delay = Duration.between(LocalDateTime.now(), task.getStartTime());
+            timer.scheduleAtFixedRate(task, delay.toMillis(), task.getLongPeriod());
+            System.out.println("Should only add once");
```

Comment:
I think that this line should be removed in the final version of the code.

Reply:
Oh yes i will